### PR TITLE
Externalise ts-lint rules

### PR DESCRIPTION
**Pull request by timjacobi**
As discussed in #109 

I wasn't entirely sure how to properly test my changes since the setup doesn't allow it to test with `npm link`. Please let me know. Thanks :)

### CODE REVIEW

File: packages/react-scripts/template/tslint.json [hunk 1]
```diff
@@ -1,99 +1,3 @@
...
-            "check-branch",
-            "check-decl",
-            "check-module",
-            "check-operator",
-            "check-separator",
-            "check-type",
-            "check-typecast"
-        ]
-    }
+    "extends": ["tslint-react", "tslint-create-react-app-typescript"]
```
Comment (wmonk) [inline packages/react-scripts/template/tslint.json]: To be honest i'd be happier to just extend from `tslint-react` and let people write their own rules? Would be interested to hear how people are actually using this file, whether they just delete their own and start again, or leave as-is.

Comment (timjacobi) [inline packages/react-scripts/template/tslint.json]: That would make things a lot simpler

Comment (makepost) [inline packages/react-scripts/template/tslint.json]: [Sample `tslint-react` config](https://github.com/palantir/tslint-react/blob/master/README.md#usage) suggests extending [tslint:latest](https://github.com/palantir/tslint/blob/master/src/configs/latest.ts)

The `latest` preset extends [tslint:recommended](https://github.com/palantir/tslint/blob/master/src/configs/recommended.ts) and adds features supported by tslint 2.4

Many reasonable defaults

Quite strict

Helps keep consistent style across projects

Angular contributors have an [issue](https://github.com/angular/angular-cli/issues/6179) and a [pull request](https://github.com/angular/angular-cli/pull/6507) considering its usage in their bootstrapping tools

**Comment by wmonk** [comment]
@timjacobi do you want to update this?

File: packages/react-scripts/template/tslint.json (hunk 1 above)
Comment (wmonk) [inline packages/react-scripts/template/tslint.json]: I would agree this is a good starting point!

**Comment by timjacobi** [comment]
I need to check with my new employer. Shouldn't be a problem though but will take a day or two. Happy to update once greenlit.

**Comment by wmonk** [comment]
Ok great! If you can't work on this, then i'm sure someone else will be able to 👍

File: packages/react-scripts/template/tslint.json (hunk 1 above)
Comment (Falieson) [inline packages/react-scripts/template/tslint.json]: I like having all the tslint config rules that someone else thought through, but some of the rules are out of date.

Project is a day or so old, here's my tslint: https://github.com/Falieson/typescript-react-asteroid-meteor/blob/master/client/tslint.json

**Comment by ianschmitz** [comment]
Are you able to continue on this PR @timjacobi? If not i don't mind helping out. 

Sounds like from the thoughts above we would provide something like `"extends": ["tslint:latest", "tslint-react"]`, and let the developer customize from there?

**Comment by wmonk** [comment]
@ianschmitz are you interested in picking this up? Closing for now.

**Pull request closed without merging**